packages/scaffold/src/partials/w3m-convert-input/index.ts

```diff
@@ -0,0 +1,226 @@
+      'Meta',
+      'Ctrl',
+      'a',
+      'c',
+      'v',
+      'ArrowLeft',
+      'ArrowRight',
+      'Tab'
+    ]
+    const isComma = event.key === ','
+    const isDot = event.key === '.'
+    const isNumericKey = event.key >= '0' && event.key <= '9'
+    const currentValue = this.value
+
+    if (!isNumericKey && !allowedKeys.includes(event.key) && !isDot && !isComma) {
+      event.preventDefault()
+    }
+
+    if (isComma || isDot) {
+      if (currentValue?.includes('.') || currentValue?.includes(',')) {
+        event.preventDefault()
+      }
+    }
+  }
+
+  private dispatchInputChangeEvent(event: InputEvent) {
+    if (!this.onSetAmount) {
+      return
+    }
+
+    const value = (event.target as HTMLInputElement).value
+    if (value === ',' || value === '.') {
+      this.onSetAmount(this.target, '0.')
+    } else if (value.endsWith(',')) {
+      this.onSetAmount(this.target, value.replace(',', '.'))
+    } else {
+      this.onSetAmount(this.target, value)
+    }
+  }
+
+  private setMaxValueToInput() {
+    this.onSetMaxValue?.(this.target, this.balance)
+  }
+
+  private templateTokenSelectButton() {
+    if (!this.token) {
+      return html` <wui-button
+        class="swap-token-button"
+        size="md"
+        variant="accentBg"
+        @click=${this.onSelectToken.bind(this)}
+      >
+        Select token
+      </wui-button>`
+    }
+
+    const tokenElement = this.token.logoURI
+      ? html`<wui-image src=${this.token.logoURI}></wui-image>`
+      : html`
+          <wui-icon-box
+            size="sm"
+            iconColor="fg-200"
+            backgroundColor="fg-300"
+            icon="networkPlaceholder"
+          ></wui-icon-box>
+        `
+
+    return html`
+      <wui-flex
+        class="swap-token-button"
+        flexDirection="column"
+        alignItems="flex-end"
+        justifyContent="center"
+        gap="xxs"
+      >
+        <button
+          size="sm"
+          variant="shade"
+          class="token-select-button"
+          @click=${this.onSelectToken.bind(this)}
+        >
+          ${tokenElement}
+          <wui-text variant="paragraph-600" color="fg-100">${this.token.symbol}</wui-text>
+        </button>
+        <wui-flex alignItems="center" gap="xxs"> ${this.tokenBalanceTemplate()} </wui-flex>
+      </wui-flex>
+    `
+  }
+
+  private tokenBalanceTemplate() {
+    const balanceValueInUSD = NumberUtil.multiply(this.balance, this.price)
+    const haveBalance = balanceValueInUSD
+      ? balanceValueInUSD?.isGreaterThan(MINIMUM_USD_VALUE_TO_CONVERT)
+      : false
+
+    return html`
+      ${haveBalance
+        ? html`<wui-text variant="small-400" color="fg-200">
+            ${UiHelperUtil.formatNumberToLocalString(this.balance, 3)}
+          </wui-text>`
+        : null}
+      ${this.target === 'sourceToken' ? this.tokenActionButtonTemplate(haveBalance) : null}
+    `
+  }
+
+  private tokenActionButtonTemplate(_haveBalance: boolean) {
+    if (_haveBalance) {
```

Comment:
why the underscore arg? That usually means unused

Reply:
It was because of the conflict with a variable used with same name in the class. But for this one, we don't actually need it. Good call, removed